src/main/scala/beam/agentsim/agents/HasTickAndTrigger.scala

```diff
@@ -21,5 +24,7 @@ trait HasTickAndTrigger {
   }
 
   def getCurrentTick: Option[Int] = _currentTick
+  def getCurrentTriggerId: Option[Long] = _currentTriggerId
+  def getCurrentTriggerIdOrGenerate: Long = _currentTriggerId.getOrElse(-rnd.nextInt(Int.MaxValue))
```

Comment:
I think that it's better not to set the triggerId here because it would unite all the subsequent messages of this actor in a single message sequence (until the new triggerID is set). But all these messages are separated.

Reply:
Agreed - the only addition is that it seems as though if it hits this point then it should have been set - and we should know where it goes empty and/or the full trace from the beginning?